token_signature.go

```diff
@@ -191,28 +117,9 @@ func verifyCertificates(root, inter, leaf *x509.Certificate) error {
 	return nil
 }
 
-// verifyPKCS7Signature verifies that the signature was produced by the leaf
-// certificate contained in the given PKCS7 struct
-func (t PKPaymentToken) verifyPKCS7Signature(p7 *C.PKCS7) error {
-	// TODO: use the Go x509 API instead of OpenSSL
-	// This code does not work for some reason:
-	// if err := leaf.CheckSignature(leaf.SignatureAlgorithm, t.signedData(),
-	//     signatureBytes); err != nil {
-	//
-	//     return errors.Wrap(err, "invalid signature")
-	// }
-
-	C.OpenSSL_add_all_algorithms_func()
-	//defer C.EVP_cleanup()
-	signedDataBio := newBIOBytes(t.signedData())
-	defer signedDataBio.Free()
-	// The PKCS7_NOVERIFY flag corresponds to verifying the chain of trust of
-	// the certificates, which should have been done before
-	r := C.PKCS7_verify(p7, nil, nil, signedDataBio.C(), nil, C.PKCS7_NOVERIFY)
-	if r != 1 {
-		return errors.Wrap(opensslErr(), "signature validation error")
-	}
-	return nil
+func (t PKPaymentToken) verifyPKCS7Signature(p7 *pkcs7.PKCS7) error {
+	p7.Content = t.signedData()
```

Comment:
may be good to put a comment here why we're doing this, I bet we'll forget it until next time someone has to dig into this code 😀